app/controllers/api/v2/accredited_provider_training_providers_controller.rb

```diff
@@ -50,7 +50,21 @@ def index
 
         providers = TrainingProviderSearch.new(provider: @provider, params: params).call
 
-        render jsonapi: providers, include: params[:include]
+        course_counts = {}
+
+        providers.map do |provider|
+          courses_count = provider
+                            .courses
+                            .where(accrediting_provider_code: @provider.provider_code,)
+                            .count
+          course_counts[provider.provider_code] = courses_count
+        end
```

Comment:
There is also an [existing method `include_course_counts`](https://github.com/DFE-Digital/teacher-training-api/blob/master/app/models/provider.rb#L157-L168) that optimises this:

```ruby
  # This is used by the providers index; it is a replacement for `.includes(:courses)`,
  # but it only fetches the counts for the associated courses. By not fetching all the
  # course objects for 1000+ providers, the db query runs much faster, and the view spends
  # less time rendering because there's less data to comb through.
  def self.include_courses_counts
    joins(
      %{
        LEFT OUTER JOIN (
          SELECT b.provider_id, COUNT(*) courses_count
          FROM course b
          WHERE b.discarded_at IS NULL
          GROUP BY b.provider_id
        ) a ON a.provider_id = provider.id
      },
    ).select("provider.*, COALESCE(a.courses_count, 0) AS included_courses_count")
  end
```

Used in [API::V2::ProvidersController#index](https://github.com/DFE-Digital/teacher-training-api/blob/master/app/controllers/api/v2/providers_controller.rb#L16).

I think this would change the shape of the response, iirc it adds the course count to the meta for each individual provider, but it'd be nice to use an existing pattern.

Reply:
I'm uncertain whether given the granularity of the end point whether the accredited_courses_count meta should be added to each individual Provider resource or this should be added at the root of the payload as described on line 64

Reply:
Leaving the new meta object in the root as discussed